lpwan-backend/src/main/java/com/cumulocity/lpwan/lns/connection/model/LpwanDeviceFilter.java

```diff
@@ -4,16 +4,14 @@
 import com.cumulocity.sdk.client.inventory.InventoryFilter;
 
 public class LpwanDeviceFilter extends InventoryFilter {
-
     @ParamSource
     private String query;
 
-    private LpwanDeviceFilter(String filterCriteria, String filterCriteriaValue) {
-        query = "$filter=c8y_LpwanDevice." + filterCriteria + " eq " +
-                String.format("'%s'", filterCriteriaValue);
+    private LpwanDeviceFilter(String query) {
+        this.query = query;
     }
 
-    public static LpwanDeviceFilter of(String filterCriteria, String filterCriteriaValue) {
-        return new LpwanDeviceFilter(filterCriteria, filterCriteriaValue);
+    public static LpwanDeviceFilter byServiceProviderAndLnsConnectionName(String serviceProvider, String lnsConnectionName) {
+        return new LpwanDeviceFilter(String.format("$filter=c8y_LpwanDevice.serviceProvider eq '%s' and c8y_LpwanDevice.lnsConnectionName eq '%s'", encode(serviceProvider), encode(lnsConnectionName)));
```

Comment:
The final variable didnt work due to the ParamSource annotation issue. Hence using the String format way.